filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -1510,7 +1509,9 @@ void OpenGLDriver::destroyTexture(Handle<HwTexture> th) {
 
     if (th) {
         GLTexture* t = handle_cast<GLTexture*>(th);
-        if (UTILS_LIKELY(t->usage & TextureUsage::SAMPLEABLE)) {
+        if (UTILS_UNLIKELY(t->target == SamplerType::SAMPLER_EXTERNAL)) {
+            mPlatform.destroyExternalImage(t);
```

Comment:
What happens on Android here?  It looks like PlatformEGL.h doesn't implement destroyExternalImage(), right? Are we not leaking the external image in that case?

Reply:
You're right, we need to make sure we delete the texture.

Can we assume that all external images are sampleable? If so, then I think we need to make sure to unbind the texture, delete any renderbuffers and detach streams (there shouldn't be either for external images). I can rearrange the deletion logic like this:

```
        if (UTILS_LIKELY(t->usage & TextureUsage::SAMPLEABLE)) {
            unbindTexture(t->gl.target, t->gl.id);
            if (UTILS_UNLIKELY(t->hwStream)) {
                detachStream(t);
            }
            if (t->gl.rb) {
                glDeleteRenderbuffers(1, &t->gl.rb);
            }
            if (UTILS_UNLIKELY(t->target == SamplerType::SAMPLER_EXTERNAL)) {
                mPlatform.destroyExternalImage(t);
            } else {
                glDeleteTextures(1, &t->gl.id);
            }
        }
```

Then EGL's destroyExternalImage(t) can call glDeleteTextures, and iOS's can do its own thing.

Reply:
Yes I think it's okay to assume they're sampleable. Maybe we need to enforce it.